postsrsd.c

```diff
@@ -173,6 +174,21 @@ static void handle_forward (srs_t *srs, FILE *fp, const char *address, const cha
       return;
     }
   }
+  /* Comment from Mark Kramer's envfrom2srs.pl:
+   * "Ok, first check whether we already have a signed SRS address;
+   * if so, just return the old address: we do not want to double-sign
+   * by accident! (Non-locally generated SRS0 addresses, by nature
+   * of the protocol, will not 'eval'; so, they will simply become
+   * SRS1 addresses. Thus, only locally generated SRS0 addresses are
+   * exempted from double-signing.)" */
+  result = srs_reverse(srs, value, sizeof(value), address);
+  if (result == SRS_SUCCESS) {
+    output = url_encode(outputbuf, sizeof(outputbuf), address);
+    fprintf (fp, "200 %s\n", output);
+    syslog (LOG_MAIL | LOG_INFO, "srs_forward: <%s> already signed", address);
+    fflush (fp);
+    return;
+  }
```

Comment:
1. You should return a "500 Already resigned" result here, to remain consistent with the rest of PostSRSd, which will only return a 200 result if a rewrite actually took place (this is important because a 200 result will end processing the Postfix canonical_maps, which should only happen on a successful rewrite).

2. I'm not exactly sure why this code block is neccessary to begin with. Is this assuming there is another program doing SRS rewrites?